### PR TITLE
Generics: Refactor TestLocalScan

### PR DESCRIPTION
**Pull request by chenjunjiedada**
This fixes #https://github.com/apache/iceberg/issues/1262.

### CODE REVIEW

**Comment by chenjunjiedada** [comment]
Hi @rdsr, would you mind to take a look?

File: data/src/test/java/org/apache/iceberg/data/TestLocalScan.java [hunk 1]
```diff
@@ -103,132 +103,100 @@ public TestLocalScan(String format) {
...
-    );
-    DataFile file21 = writeFile(sharedTableLocation, format.addExtension("file-21"), file2FirstSnapshotRecords);
-
-    this.file3FirstSnapshotRecords = Lists.newArrayList(
-        record.copy(ImmutableMap.of("id", 24L, "data", "cloud")),
-        record.copy(ImmutableMap.of("id", 25L, "data", "zen")),
-        record.copy(ImmutableMap.of("id", 26L, "data", "sky"))
-    );
-    DataFile file31 = writeFile(sharedTableLocation, format.addExtension("file-31"), file3FirstSnapshotRecords);
+    DataFile file12 = writeFile(sharedTableLocation, format.addExtension("file-11"), file1SecondSnapshotRecords);
```
Comment (edgarRd) [inline data/src/test/java/org/apache/iceberg/data/TestLocalScan.java:158]: It's a bit confusing that the name of the `DataFile` variable is different from the extension name, e.g. in this case `DataFile` is `file12` and extension `file-11`. Is there a reason for the rename from how it was before?

Comment (edgarRd) [reply data/src/test/java/org/apache/iceberg/data/TestLocalScan.java:158]: I think this renaming produces some changes down below too.

File: data/src/test/java/org/apache/iceberg/data/TestLocalScan.java [hunk 2]
```diff
@@ -103,132 +103,100 @@ public TestLocalScan(String format) {
...
-  private List<Record> file1FirstSnapshotRecords = null;
-  private List<Record> file2FirstSnapshotRecords = null;
-  private List<Record> file3FirstSnapshotRecords = null;
-  private List<Record> file1SecondSnapshotRecords = null;
-  private List<Record> file2SecondSnapshotRecords = null;
-  private List<Record> file3SecondSnapshotRecords = null;
 
+  private final Record genericRecord = GenericRecord.create(SCHEMA);
+
+  private final List<Record> file1FirstRecords =  Lists.newArrayList(
```
Comment (edgarRd) [inline data/src/test/java/org/apache/iceberg/data/TestLocalScan.java:109]: Looks like the goal is to make these immutable, have you considered using `ImmutableList` too?

File: data/src/test/java/org/apache/iceberg/data/TestLocalScan.java (hunk 1 above)
Comment (chenjunjiedada) [reply data/src/test/java/org/apache/iceberg/data/TestLocalScan.java]: Nice catch, fixed the extension. 

The first snapshot is actually consisted of `file1FirstRecords`, `file2FirstRecords` and `file3FirstRecords`, so I renamed this to match the actual snapshot order.

File: data/src/test/java/org/apache/iceberg/data/TestLocalScan.java (hunk 2 above)
Comment (chenjunjiedada) [reply data/src/test/java/org/apache/iceberg/data/TestLocalScan.java]: Good idea. Applied.

**Comment by chenjunjiedada** [comment]
@edgarRd , Thanks for your comments! @rdsr, would you please help to take a look as well?

**Comment by rdsr** [comment]
Thanks @chenjunjiedada . Looking into it now.

**Pull request merged**